src/gui/editors/TimeLineWidget.cpp

```diff
@@ -273,9 +273,9 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 	for( int i = 0; x + i * m_ppb < width(); ++i )
 	{
 		++barNumber;
-		if( ( barNumber - 1 ) %
-			qMax( 1, qRound( 1.0f / 3.0f *
-				TimePos::ticksPerBar() / m_ppb ) ) == 0 )
+		// round m_ppb to the nearest power of two
+		float ppbP2 = pow(2, round(log2(m_ppb)));
+		if ((barNumber - 1) % qMax(1, qRound(1.0f / 3.0f * TimePos::ticksPerBar() / ppbP2)) == 0)
```

Comment:
```suggestion
		const float ppbP2 = std::pow(2, std::lround(std::log2(m_ppb)));
		if ((barNumber - 1) % std::max(1l, std::lround((1.0f / 3.0f) * TimePos::ticksPerBar() / ppbP2)) == 0)
```
This removes Qt functions and fixes a compile error on my machine. You'll need to add `#include <cmath>` at the top of this file though. Maybe you could fix the problems with the [include order](https://github.com/LMMS/lmms/wiki/Coding-Conventions#include-order) while you're at it.

Reply:
ok